components/Menu/Menu.js

```diff
@@ -128,23 +134,43 @@ export default function Menu({
               disabled={!session}
             />
 
-            <StyledMenuLogin
-              onClick={() => {
-                changeSubmenuMode("default");
-                changeFlashcardSelection("all");
-                handleToggleMenu();
-              }}
-            >
+            <StyledMenuLogin>
               <Login
                 position="menu"
                 changeShowLogOutDialog={changeShowLogOutDialog}
-                showLogOutDialog={showLogOutDialog}
+                changeSubmenuMode={changeSubmenuMode}
+                changeFlashcardSelection={changeFlashcardSelection}
+                handleToggleMenu={handleToggleMenu}
               />
             </StyledMenuLogin>
           </StyledNavigationList>
         </StyledNavigation>
       )}
       {isMenu && <StyledCover />}
+
+      {showLogOutDialog && (
+        <StyledOutgreyContainer>
+          <ModalContainer>
+            <ModalQuestion>Do you really want to log out?</ModalQuestion>
+            <ButtonWrapper>
+              <RegularButton
+                onClick={() => handleLogout()}
+                type="button"
+                variant="warning"
+              >
+                Yes
+              </RegularButton>
+              <RegularButton
+                onClick={() => changeShowLogOutDialog(false)}
+                type="button"
+                variant="confirm"
+              >
+                Cancel
+              </RegularButton>
+            </ButtonWrapper>
+          </ModalContainer>
+        </StyledOutgreyContainer>
+      )}
```

Comment:
In a later ticket you could unify this with the other modal your college has created and create a dedicated component for it.

Reply:
Yes, we've already thought about that. You mentioned that there is an extra modal level, so to speak. But you wouldn't recommend that we implement that?